src/inmanta/warnings.py

```diff
@@ -142,16 +142,18 @@ def _showwarning(
             :param file: The file to write the warning to. Defaults to stderr.
             :param line: Required for compatibility but will be ignored.
         """
-        # implementation based on warnings._showwarnmsg_impl
-        if file is None:
-            file = sys.stderr
-            if file is None:
-                return
-        text: str = "WARNING %s: %s\n" % (category.__name__, message)
-        try:
-            file.write(text)
-        except OSError:
-            pass
+        # implementation based on warnings._showwarnmsg_impl and logging._showwarning
+        text: str = "%s: %s\n" % (category.__name__, message)
+        if file is not None:
+            try:
+                file.write(text)
+            except OSError:
+                pass
+        else:
+            logger = logging.getLogger("inmanta.warnings")
+            if not logger.handlers:
+                logger.addHandler(logging.NullHandler())
```

Comment:
Why?

Reply:
Pass it through the logger. It's not pretty but I'm afraid that's just the state of the python warnings module. The logging module overwrites warnings.showwarnings if you toggle captureWarnings so as far as I see this is the best way to pair a custom warnings formatting with the logger.

Reply:
Sorry, I have to be more precise: 
Why attach the NullHandler?

Reply:
It was in the [logging _showwarning implementation](url) and I assumed a logging requires a handler. I could investigate this further.

Reply:
relevant documentation: https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library

Reply:
It seems to prevent warnings to be redirected to stderr when used in a library for an application which does not use logging itself. I'm still a bit confused about why it's the behaviour for `logging.captureWarnings()` but it doesn't really seem relevant for our use case so I think I should remove it. Do you agree?

Reply:
I do

Reply:
Fixed. I also updated the implementation to not reformat non-Inmanta warnings, though they are currently ignored.